lightning/src/ln/channelmanager.rs

```diff
@@ -98,6 +98,12 @@ use core::ops::Deref;
 pub use crate::ln::outbound_payment::{PaymentSendFailure, ProbeSendFailure, Retry, RetryableSendFailure, RecipientOnionFields};
 use crate::ln::script::ShutdownScript;
 
+#[cfg(not(c_bindings))]
+type OfferBuilder<'a> =
+	crate::offers::offer::OfferBuilder<'a, crate::offers::offer::DerivedMetadata, secp256k1::All>;
+#[cfg(c_bindings)]
+type OfferBuilder<'a> = crate::offers::offer::OfferWithDerivedMetadataBuilder<'a>;
```

Comment:
This is for the return type of `ChannelManager::create_offer_builder`. The compiler doesn't like having a `#[cfg()]` in the middle of a return type apparently.

Alternatives seem to be either:
- define a macro to avoid duplicating the code and docs (though the docs reference the alias).
- duplicate the code and docs

The code is short enough and also contains `#[cfg()]`, so might be best just to duplicate it.

Reply:
Last push uses a macro. I'll need to do the same for `create_refund_builder` and didn't want to add more lines to `channelmanager.rs`. A type alias would be better if possible, IMO.